src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -200,26 +202,38 @@ class Peer(
         }
         launch {
             finalWallet.walletStateFlow
-                .distinctUntilChangedBy { Pair(it.unconfirmedBalance, it.confirmedBalance) }
+                .distinctUntilChangedBy { it.totalBalance }
                 .collect { wallet ->
-                    logger.info { "${wallet.totalBalance} available on final wallet (${wallet.unconfirmedBalance} unconfirmed)" }
+                    logger.info { "${wallet.totalBalance} available on final wallet with ${wallet.utxos.size} utxos" }
                 }
         }
         launch {
             swapInWallet.walletStateFlow
                 .filter { it.consistent }
                 .fold(emptySet<WalletState.Utxo>()) { reservedUtxos, wallet ->
-                    // reservedUtxos are part of a previously issued RequestChannelOpen command
-                    val availableWallet = wallet.minus(reservedUtxos)
-                    val balance = availableWallet.confirmedBalance
-                    logger.info { "swap-in wallet balance: confirmed=$balance, unconfirmed=${availableWallet.unconfirmedBalance}" }
-                    if (balance > 0.sat) {
-                        logger.info { "swap-in wallet: requesting channel using confirmed balance: $balance" }
-                        input.send(RequestChannelOpen(Lightning.randomBytes32(), availableWallet))
-                        reservedUtxos.union(availableWallet.confirmedUtxos)
-                    } else {
-                        reservedUtxos
-                    }.intersect(wallet.utxos.toSet()) // drop utxos no longer in wallet
+                    when (val currentBlockHeight = currentTipFlow.value?.first) {
+                        null -> {
+                            logger.warning { "cannot process wallet state: current block height not available" }
+                            reservedUtxos
+                        }
+                        else -> {
+                            // reservedUtxos are part of a previously issued RequestChannelOpen command
+                            val availableWallet = wallet.minus(reservedUtxos).withConfirmations(currentBlockHeight, walletParams.swapInConfirmations)
+                            logger.info { "swap-in wallet balance: deeplyConfirmed=${availableWallet.deeplyConfirmed.balance}, weaklyConfirmed=${availableWallet.weaklyConfirmed.balance}, unconfirmed=${availableWallet.unconfirmed.balance}" }
+                            val utxos = when {
+                                // When migrating from the legacy android app, we use all utxos, even unconfirmed ones.
+                                isMigrationFromLegacyApp -> availableWallet.all
```

Comment:
I think logging differently based on `isMigrationFromLegacyApp` like I did on my branch could help debugging.

Reply:
When we get tired of seeing `(migration=false)` in the logs we'll remove that special case.